src/main/java/org/opensearch/sdk/BaseExtensionRestHandler.java

```diff
@@ -127,4 +135,58 @@ protected ExtensionRestResponse createJsonResponse(RestRequest request, RestStat
             return new ExtensionRestResponse(request, status, fieldName + ": " + responseStr);
         }
     }
+
+    /**
+     * Returns a String message of the detail of any unrecognized error occurred
+     *
+     * @param request The request that caused the exception
+     * @param invalids invalid strings
+     * @param candidates candidates involved
+     * @param detail
+     * @return an String that contains the message.
+     */
+    public static final String unrecognized(RestRequest request, Set<String> invalids, Set<String> candidates, String detail) {
+        StringBuilder message = new StringBuilder(
+            String.format(Locale.ROOT, "request [%s] contains unrecognized %s%s: ", request.path(), detail, invalids.size() > 1 ? "s" : "")
+        );
+        boolean first = true;
+
+        for (Iterator var7 = invalids.iterator(); var7.hasNext(); first = false) {
```

Comment:
Should be 
```suggestion
        for (final String invalid : invalids) {
```

Reply:
The variable first is always initialized as false in every iteration. If I accept you commit suggestion then I have to add another line which will initialized first as false for every iteration.  Therefore my suggestion would be to not change anything.

Reply:
See the next comment. Copy/paste the method exactly. That will override this change

Reply:
Got it. Pushed the updated changes.
Thanks